packages/frontend/core/src/modules/share-menu/view/share-menu/share-menu.tsx

```diff
@@ -60,9 +62,21 @@ export const ShareMenuContent = (props: ShareMenuProps) => {
   const subscription = useLiveData(
     workspaceSubscriptionService.subscription.subscription$
   );
-  const hittingPaywall =
-    (!subscription && !isSelfhosted) ||
-    (subscription && subscription.plan === SubscriptionPlan.Free);
+  const workspaceQuotaService = useService(WorkspaceQuotaService);
+  const quota = useLiveData(workspaceQuotaService.quota.quota$);
+  const hittingPaywall = useMemo(() => {
+    if (isSelfhosted) {
+      return false;
+    }
+    if (subscription) {
+      return subscription.plan === SubscriptionPlan.Free;
+    }
+    if (quota) {
+      const { name } = quota;
+      return name.toLowerCase() === SubscriptionPlan.Free.toLowerCase();
+    }
+    return true;
+  }, [isSelfhosted, subscription, quota]);
 
```

Comment:
`workspaceQuotaService` needs explicit revalidate

```suggestion
useEffect(() => {
    workspaceQuotaService.quota.revalidate();
  }, [
    workspaceQuotaService
  ]);
```